src/nvme/fabrics.c

```diff
@@ -473,6 +470,7 @@ static int build_options(nvme_host_t h, nvme_ctrl_t c, char **argstr)
 	     add_int_argument(argstr, "fast_io_fail_tmo",
 			      cfg->fast_io_fail_tmo, false)) ||
 	    (strcmp(transport, "loop") &&
+	     cfg->tos != -1 &&
```

Comment:
That is an unrelated fix, and the underlying issue (of tos not being initialized correctly) has already been handled in PR #110.

Reply:
I had to add this because tos default value is -1; before this patch, tos was filtered out by add_int_argument(), but now it's passed to the kernel and the latter returns an error.

Reply:
But isn't the error rather that 'tos' is not supported for loop?
IE shouldn't we exclude 'tos' entirely for 'loop'?

Reply:
I'm not sure I follow you here,  "tos" is already excluded when the transport is equal to "loop", it's what that strcmp() instruction is supposed to do.

Reply:
But my point was that this fix is for 'ctrl_loss_tmo' handling, not 'tos' handling.
Yet this patch introduces changes to 'tos' handling.